src/equations/compressible_navier_stokes_2d.jl

```diff
@@ -0,0 +1,239 @@
+Further, the nondimensionalization takes the density-temperature-sound speed as
+the principle quantities such that
+rho_inf = 1.0
+T_ref = 1.0
+c_inf = 1.0
+p_inf = 1.0 / gamma
+u_inf = Ma_inf
+R = 1.0 / gamma
+
+Other normalization strategies exist, see the reference below for details.
+- Marc Montagnac (2013)
+  Variable Normalization (nondimensionalization and scaling) for Navier-Stokes
+  equations: a practical guide
+  [CERFACS Technical report](https://www.cerfacs.fr/~montagna/TR-CFD-13-77.pdf)
+The scaling used herein is Section 4.5 of the reference.
+
+In two spatial dimensions we require gradients for three quantities, e.g.,
+primitive quantities
+  grad(u), grad(v), and grad(T)
+or the entropy variables
+  grad(w_2), grad(w_3), grad(w_4)
+where
+  w_2 = rho v_1 / p, w_3 = rho v_2 / p, w_4 = -rho / p
+"""
+# TODO:
+# 1) For now I save gamma and inv(gamma-1) again, but we could potentially reuse them from
+#    the Euler equations
+# 2) Add more here and probably some equations
+struct CompressibleNavierStokes2D{RealT<:Real, E} <: AbstractCompressibleNavierStokesEquations{2, 3}
+  gamma::RealT               # ratio of specific heats
+  inv_gamma_minus_one::RealT # = inv(gamma - 1); can be used to write slow divisions as fast multiplications
+  Re::RealT                  # Reynolds number
+  Pr::RealT                  # Prandtl number
+  Ma_inf::RealT              # free-stream Mach number
+  kappa::RealT               # thermal diffusivity for Fick's law
+
+  p_inf::RealT               # free-stream pressure
+  u_inf::RealT               # free-stream velocity
+  R::RealT                   # gas constant (depends on nondimensional scaling!)
+
+  equations::E               # CompressibleEulerEquations2D
+end
+
+function CompressibleNavierStokes2D(gamma, Reynolds, Prandtl, Mach_freestream, kappa, equations)
+  γ, inv_gamma_minus_one = promote(gamma, inv(gamma - 1))
+
+  # From the nondimensionalization discussed above set the remaining free-stream
+  # quantities
+  p_inf = 1.0 / γ
+  u_inf = Mach_freestream
+  R     = 1.0 / γ
+  CompressibleNavierStokes2D{typeof(γ),typeof(equations)}(γ, inv_gamma_minus_one,
+                                                          Reynolds, Prandtl, Mach_freestream,
+                                                          kappa, p_inf, u_inf, R,
+                                                          equations)
+end
+
+
+# I was not sure what to do here to allow flexibility of selecting primitive or entropy
+# gradient variables. I see that `transform_variables!` just copies data at the moment.
+
+# This is the flexibility a user should have to select the different gradient variable types
+# varnames(::typeof(cons2prim)   , ::CompressibleNavierStokes2D) = ("v1", "v2", "T")
+# varnames(::typeof(cons2entropy), ::CompressibleNavierStokes2D) = ("w2", "w3", "w4")
+
+varnames(variable_mapping, equations_parabolic::CompressibleNavierStokes2D) =
+  varnames(variable_mapping, equations_parabolic.equations)
```

Comment:
Ah, okay. I got confused then. In my other codes I carry around an `nvars` for the conservative variables and a separate `nvars_grad` for the number of gradient variables.

Reply:
From my end it would be helpful to (somewhere) have the constructor for the parabolic equations output what are the gradient variables for the user. What I mean is that the equations constructors would report something like the following to the screen 
```
┌──────────────────────────────────────────────────────────────────────────────────────────────────┐
│ CompressibleEulerEquations2D                                                                     │
│ ════════════════════════════                                                                     │
│ #variables: ………………………………………………… 4                                                                │
│ │ variable 1: …………………………………………… rho                                                              │
│ │ variable 2: …………………………………………… rho_v1                                                           │
│ │ variable 3: …………………………………………… rho_v2                                                           │
│ │ variable 4: …………………………………………… rho_e                                                            │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
┌──────────────────────────────────────────────────────────────────────────────────────────────────┐
│ CompressibleNavierStokes2D                                                                       │
│ ══════════════════════════                                                                       │
│ gradient variables: …………………………… primitive                                                        │
│ #variables: ………………………………………………… 3                                                                │
│ │ variable 1: …………………………………………… v1                                                               │
│ │ variable 2: …………………………………………… v2                                                               │
│ │ variable 3: …………………………………………… T                                                                │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
```

Reply:
Yeah, this sounds reasonable. Could you please add this to our issue https://github.com/trixi-framework/Trixi.jl/issues/1147 tracking parabolic stuff?